netlify/functions/feedback-management.js

```diff
@@ -40,67 +39,112 @@ const awsLambdaReceiver = new AwsLambdaReceiver({
 const app = new App({
   token: process.env.SLACK_BOT_TOKEN,
   receiver: awsLambdaReceiver,
+  signingSecret: process.env.SLACK_SIGNING_SECRET,
 });
 
-// Define slack actions to answer
-
-app.shortcut('delete_action', async ({ ack, body, client }) => {
-  await ack();
-  await client.chat.delete({
-    channel: body.channel.id,
-    ts: body.message_ts,
-    as_user: true,
-    token: process.env.SLACK_BOT_TOKEN,
-  });
-});
+/**
+ * Helper function to retrieve data from the message
+ */
+function getQuoteAndLinks(message) {
+  const text = message?.blocks?.[0]?.text?.text;
 
-app.shortcut('save_message', async ({ ack, body, client }) => {
-  await ack();
-  const {
-    user: { username },
-    channel: { id: channelId },
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    message_ts,
-    message,
-  } = body;
-  const elements = message?.blocks?.[0]?.elements;
-
-  const quote = message?.text
+  const quote = text
     .split('\n\nsent from')[0]
     .split('\n\n')
     .slice(1)
     .join('\n\n')
     .replace(/&gt;/g, '');
 
-  const links = elements[2].elements
-    .filter((element) => element.type === 'link')
-    .map((element) => element.url);
-
-  const googleAuth = new JWT({
-    email: 'service-account-804@docs-feedbacks.iam.gserviceaccount.com',
-    key: process.env.G_SHEET_TOKEN.replace(/\\n/g, '\n'),
-    scopes: ['https://www.googleapis.com/auth/spreadsheets'],
-  });
-  const service = sheets({ version: 'v4', auth: googleAuth });
-
-  await service.spreadsheets.values.append({
-    spreadsheetId: spreadSheetsIds.forLater,
-    range: 'Sheet1!A:D',
-    valueInputOption: 'USER_ENTERED',
-    resource: {
-      values: [[username, quote, links[0] ?? '', links[1] ?? '']],
-    },
-  });
-  await client.chat.postMessage({
-    channel: channelId,
-    thread_ts: message_ts,
-    as_user: true,
-    text: `Saved in <https://docs.google.com/spreadsheets/d/${spreadSheetsIds.forLater}/>`,
-  });
+  const links = text
+    .split('\n\nsent from <')[1]
+    .split(' (from section <')
+    .map((linkSection) => {
+      const splited = linkSection.split('/>');
+      if (splited.length === 2) {
+        // Ensure there is a `/>` which means we are facing a link
+        return splited[0];
+      }
+      return '';
+    })
+    .filter((link) => link !== '');
+  return { quote, links };
+}
+// Define slack actions to answer
+
+app.action('delete_action', async ({ ack, body, client, logger }) => {
+  try {
+    await ack();
+
+    const {
+      user: { username },
+      channel: { id: channelId },
+      message,
+    } = body;
+
+    const { quote, links } = getQuoteAndLinks(message);
```

Comment:
Just a suggestion, but I quickly skimmed [this doc](https://api.slack.com/reference/block-kit/block-elements#button__fields)

I may be misunderstanding it, but it looks like you can specify a `value` to pass along as payload when clicking the delete button. If so, that could offer a more structured way of passing along this information?

Reply:
I did not use it because it only accepts a string and I need multiple values. 

But now I see it could be used with `JSON.stringify({ quote, links })` and `JSON.parse`